### PR TITLE
update tooISOhistorical to handle bilateral data

### PR DESCRIPTION
**Pull request by caviddhen**
An addition to toolISOhistorical  where if 2 sub-dimensions in spatial dim are identified it runs a historical disaggregation compatible with bilateral data. Namely, there may be differing transitions in each sub-dimension, and the sub-dimensions need to be explicitly specified so as not to confuse the process.

### CODE REVIEW

**Review by pfuehrlich-pik — CHANGES_REQUESTED** [review]
I did not read the code yet, please fix the indentation first. In RStudio you can select all (ctrl + a) and then click `Code > Reindent Lines` (ctrl + i).

File: R/toolISOhistorical.R
Comment (pfuehrlich-pik) [inline R/toolISOhistorical.R:119]: The same condition is checked again in line 178, so this block could be moved there, right?

File: R/toolISOhistorical.R
Comment (pfuehrlich-pik) [inline R/toolISOhistorical.R:191]: You could consider this:
```suggestion
      toISOmissing <- !all(a$toISO %in% getItems(m, dim = 1.1))
```
I find it a bit nicer, but that's personal preference of course.

File: R/toolISOhistorical.R
Comment (pfuehrlich-pik) [inline R/toolISOhistorical.R:182]:
```suggestion
    # store original set names for later renaming, but for now rename for easy list calls later
    origDim1 <- getSets(m)[1]
    origDim2 <- getSets(m)[2]
```

File: R/toolISOhistorical.R
Comment (pfuehrlich-pik) [inline R/toolISOhistorical.R:201]:
```suggestion
        vcat(1, "Data was newly created and given equal weight for the following new countries in dim 1.1: ",
```

File: R/toolISOhistorical.R
Comment (pfuehrlich-pik) [inline R/toolISOhistorical.R:205]:
```suggestion
      # time span where the data need to be adjusted
```

File: R/toolISOhistorical.R
Comment (pfuehrlich-pik) [inline R/toolISOhistorical.R:232]: What is "Weight 1" here?

**Review by pfuehrlich-pik — CHANGES_REQUESTED** [review]
Looks good, but I added some minor notes and style suggestions. The code duplication (`# do the same thing but for the dim 1.2`) is a problem though, please use a loop or a function to get rid of the duplication.

File: R/toolISOhistorical.R
Comment (pfuehrlich-pik) [inline R/toolISOhistorical.R:160]:
```suggestion
      if (overwrite) {
```

File: R/toolISOhistorical.R
Comment (pfuehrlich-pik) [inline R/toolISOhistorical.R:223]: How is this case handled?

File: R/toolISOhistorical.R
Comment (pfuehrlich-pik) [inline R/toolISOhistorical.R:258]: Please fix the indentation here. If the line is too long with proper indentation I'd suggest to pull out the toISO part: Assign a variable `toISO` in the line before and just set `toISO = toISO` when creating the data.frame.
In case you want to use the RStudio auto-indentation fix again, please run `lucode2::autoFormat()` afterwards to remove trailing whitespace which RStudio introduces when fixing indentation.

File: R/toolISOhistorical.R
Comment (pfuehrlich-pik) [inline R/toolISOhistorical.R:281]: These `# nolint` are because of line length right? I'd strongly suggest to split this up into multiple statements, I find it very hard to read and understand it like this.

File: R/toolISOhistorical.R
Comment (pfuehrlich-pik) [inline R/toolISOhistorical.R:297]: Same here, please split this up. Also, please don't use `# nolint` in these cases, these extra long lines are hard to read, which is exactly why the linter warning exists :smile:

File: R/toolISOhistorical.R
Comment (pfuehrlich-pik) [inline R/toolISOhistorical.R:313]:
```suggestion
      if (overwrite) {
```

File: R/toolISOhistorical.R
Comment (pfuehrlich-pik) [inline R/toolISOhistorical.R:311]:
```suggestion
      mTr <- .histDim(m, tr = a, mainDim = mainDim, dims = dims)
```

File: R/toolISOhistorical.R
Comment (pfuehrlich-pik) [inline R/toolISOhistorical.R:326]: I'd suggest to split it up like so:
```suggestion
        a <- m[getItems(mTr, dim = 1), getYears(mTr), ]
        a[is.na(a)] <- mTr[is.na(a)]
```
The variable `a` should be renamed to something helpful though.

File: R/toolISOhistorical.R
Comment (pfuehrlich-pik) [inline R/toolISOhistorical.R:335]:
```suggestion
      mTr <- .histDim(m, tr = a, mainDim = mainDim, dims = dims)
```

File: R/toolISOhistorical.R
Comment (pfuehrlich-pik) [inline R/toolISOhistorical.R:337]:
```suggestion
      if (overwrite) {
```

File: R/toolISOhistorical.R
Comment (pfuehrlich-pik) [inline R/toolISOhistorical.R:350]: I've seen that line before, can we get rid of this code duplication?

**Review by pfuehrlich-pik — CHANGES_REQUESTED** [review]
Looks good! Please split up these super long lines, then we can merge this.

File: R/toolISOhistorical.R
Comment (caviddhen) [reply R/toolISOhistorical.R:350]: i would rather opt to keep this one here... as it's the 2 separate loops doing filling in...

File: R/toolISOhistorical.R
Comment (pfuehrlich-pik) [inline R/toolISOhistorical.R:224]:
```suggestion
          # these countries are added below if missing, and given weight 0
```

Comment (pfuehrlich-pik) [reply R/toolISOhistorical.R:224]: > ...leading to weights that don't sum to 1

Giving weight 0 won't help with summing up to 1, right?

File: R/toolISOhistorical.R
Comment (pfuehrlich-pik) [reply R/toolISOhistorical.R:250]: Can you clarify this?

File: R/toolISOhistorical.R
Comment (pfuehrlich-pik) [inline R/toolISOhistorical.R:290]:
```suggestion
                             weight = wTr + weight2, # add small weight to disaggregate properly
```
`# nolint` should not be necessary anymore, right?

**Comment by tscheypidi** [comment]
@caviddhen Can you please add some tests testing the new code you added? This should avoid the overall test coverage to drop by 4.5%

**Comment by pfuehrlich-pik** [comment]
The last commit really helped making the code more readable, thanks!

File: R/toolISOhistorical.R
Comment (caviddhen) [reply R/toolISOhistorical.R]: Weight 1 no longer relevant now that there is an internal function

**Review by pfuehrlich-pik — APPROVED** [review]
Looks good to me :+1:

**Comment by caviddhen** [comment]
> Looks good to me 👍

Thanks Pascal! Jan asked me to write some tests and while doing so i noticed some small things to adjust but almost all content is still the same (and no long lines). I'll wait for Jan's approval and then merge. Thanks for all the help here!

**Review by tscheypidi — APPROVED** [review]
thanks for adding the tests

**Pull request merged**
